File: monitoreo/src/main/java/models/entidades/organizaciones/entidad/TipoEntidad.java
package models.entidades.organizaciones.entidad;

import models.entidades.organizaciones.Establecimiento;
import lombok.Getter;
import lombok.Setter;

import javax.persistence.*;

@Entity
@Table(name = "tipo_entidad")
public class TipoEntidad {

  @Id
  @GeneratedValue(strategy = GenerationType.IDENTITY)
  @Getter@Setter
  private int id;

  @Getter @Setter
  @Column(name = "nombre")
  private String nombre;

  @Getter @Setter
  @Column(name = "prefijo_entidad")
  private String prefijoEntidad;

  @Getter @Setter
  @Column(name = "prefijo_establecimiento")
  private String prefijoEstablecimiento;

  public TipoEntidad (String nombre, String prefijoEntidad, String prefijoEstablecimiento) {
    this.nombre = nombre;
    this.prefijoEntidad = prefijoEntidad;
    this.prefijoEstablecimiento = prefijoEstablecimiento;
  }

  public TipoEntidad(){
  }

  public String getNombreCompletoEntidad(Entidad entidad) {

    return this.prefijoEntidad + " " + entidad.getNombre();
  }

  public String getNombreCompletoEstablecimiento(Establecimiento establecimiento) {

    return this.prefijoEstablecimiento + " " + establecimiento.getNombre();
  }

}

File: MonitoreoCNET/MonitoreoCNET/Models/Organizaciones/TipoEntidad.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace MonitoreoCNET.Models.Organizaciones
{
    [Table("tipo_entidad")]
    public class TipoEntidad
    {
        [Key]
        [DatabaseGenerated(DatabaseGeneratedOption.Identity)]
        public int Id { get; set; }

        [Column("nombre")]
        public string Nombre { get; set; }

        [Column("prefijo_entidad")]
        public string PrefijoEntidad { get; set; }

        [Column("prefijo_establecimiento")]
        public string PrefijoEstablecimiento { get; set; }

        public TipoEntidad(string nombre, string prefijoEntidad, string prefijoEstablecimiento)
        {
            Nombre = nombre;
            PrefijoEntidad = prefijoEntidad;
            PrefijoEstablecimiento = prefijoEstablecimiento;
        }

        public TipoEntidad()
        {
        }

        public string GetNombreCompletoEntidad(Entidad entidad)
        {
            return PrefijoEntidad + " " + entidad.Nombre;
        }

        public string GetNombreCompletoEstablecimiento(Establecimiento establecimiento)
        {
            return PrefijoEstablecimiento + " " + establecimiento.Nombre;
        }
    }
}